### PR TITLE
Gce parametrization and entities refactor

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Adding parametrization for puppet/non_puppet based scenarios and entities refactor for GCE module

Test results **API**:
```
pytest tests/foreman/api/test_computeresource_gce.py
================================================================================================================================== test session starts ===================================================================================================================================
platform linux -- Python 3.11.1, pytest-7.2.2, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.10.0, reportportal-5.1.6, ibutsu-2.2.4, xdist-3.2.1, cov-3.0.0
collected 11 items                                                                                                                                                                                                                                                                       

tests/foreman/api/test_computeresource_gce.py ...........                                                                                                                                                                                                                          [100%]


====================================================================================================================== 11 passed, 57 warnings in 741.56s (0:12:21) =======================================================================================================================
```

Test results **UI**:
```
pytest tests/foreman/ui/test_computeresource_gce.py
================================================================================================================================== test session starts ===================================================================================================================================
platform linux -- Python 3.11.1, pytest-7.2.2, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.10.0, reportportal-5.1.6, ibutsu-2.2.4, xdist-3.2.1, cov-3.0.0
collected 3 items                                                                                                                                                                                                                                                                        

tests/foreman/ui/test_computeresource_gce.py ...                                                                                                                                                                                                                                   [100%]

====================================================================================================================== 3 passed, 44 warnings in 1275.47s (0:21:15) =======================================================================================================================
```

### CODE REVIEW

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

File: pytest_fixtures/component/provision_gce.py [hunk 1]
```diff
@@ -5,40 +5,69 @@
...
-        json.dump(cert, f)
-    session_target_sat.put(gce_cert_file, settings.gce.cert_path)
-    if session_target_sat.execute(f'[ -f {settings.gce.cert_path} ]').status != 0:
-        raise GCECertNotFoundError(
-            f"The GCE certificate in path {settings.gce.cert_path} is not found in satellite."
-        )
-    return cert
+def sat_gce(request, session_puppet_enabled_sat, session_target_sat):
+    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
+    yield hosts[request.param]
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_gce.py:20]: I see, we've some tests which we marking only for sat, so would suggest using getattr where we make sat default and you don't need to mark parametrize if you don't want to run on both sat
```suggestion
    hosts = {'sat': session_target_sat, 'puppet_sat': session_puppet_enabled_sat}
    yield hosts[getattr(request, 'param', 'sat')]
```

Comment (adarshdubey-star) [reply pytest_fixtures/component/provision_gce.py]: Works well @Gauravtalreja1

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by adarshdubey-star** [comment]
I have merged one-test scenario of external IP assertion into `test_positive_gce_host_provisioned` as we discussed @Gauravtalreja1 @shweta83 @shubhamsg199

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

File: tests/foreman/ui/test_computeresource_gce.py [hunk 2]
```diff
@@ -57,22 +56,23 @@ def test_positive_default_end_to_end_with_custom_profile(
...
         name=gen_string('alpha', 15),
         url=settings.http_proxy.auth_proxy_url,
         username=settings.http_proxy.username,
         password=settings.http_proxy.password,
-        organization=[module_org.id],
-        location=[module_location.id],
+        organization=[sat_gce_org.id],
+        location=[sat_gce_loc.id],
     ).create()
     with session:
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_computeresource_gce.py:73]: If we think to run any of this tests against puppet_sat, or we can be explicicit by using ui_session 
```suggestion
    with sat_gce.ui_session() as session:
```

**Review by Gauravtalreja1 — APPROVED** [review]
Ack, with non-blocking comments

File: tests/foreman/ui/test_computeresource_gce.py [hunk 3]
```diff
@@ -151,12 +151,12 @@ def test_positive_default_end_to_end_with_custom_profile(
 @pytest.mark.parametrize('setting_update', ['destroy_vm_on_host_delete'], indirect=True)
 def test_positive_gce_provision_end_to_end(
     session,
-    target_sat,
```
[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply tests/foreman/ui/test_computeresource_gce.py:154]: So with the API, we have marked the puppet_sat/sat for the same scenario as e2e, so we need not do that again here.

File: tests/foreman/ui/test_computeresource_gce.py (hunk 2 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_computeresource_gce.py:73]: I think the session is generic and should not be doing anything with the sat_object being used @omkarkhatavkar @shweta83  ?

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

File: tests/foreman/ui/test_computeresource_gce.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_computeresource_gce.py:154]: yes, but we might need any tests to run in UI as well to confirm if it's going well with the puppet plugin installed

[the remark this answers is not in the document]
Comment (shweta83) [reply tests/foreman/ui/test_computeresource_gce.py:154]: We anyway don't run any ui test case in stream so we can skip it.

[the remark this answers is not in the document]
Comment (adarshdubey-star) [reply tests/foreman/ui/test_computeresource_gce.py:154]: Updated

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_computeresource_gce.py tests/foreman/ui/test_computeresource_gce.py
pod_resources_size: extra_extra_large

**Review by shweta83 — APPROVED** [review]
Refactoring and cleanup nicely done. Great work @adarshdubey-star

**Review by omkarkhatavkar — CHANGES_REQUESTED** [review]
https://github.com/SatelliteQE/robottelo/blob/1f7625400245b4b02b4d52bf79d4e32e90eaba
blocking: we have to update the bc/tests/upgrades/test_host.py#L63 so that it will using a single satellite for the upgrade tests.

**Comment by adarshdubey-star** [comment]
> https://github.com/SatelliteQE/robottelo/blob/1f7625400245b4b02b4d52bf79d4e32e90eababc/tests/upgrades/test_host.py#L63
> 
> 
> blocking: we have to update the so that it will using a single satellite for the upgrade tests.

On it

**Comment by omkarkhatavkar** [comment]
> > https://github.com/SatelliteQE/robottelo/blob/1f7625400245b4b02b4d52bf79d4e32e90eababc/tests/upgrades/test_host.py#L63
> > 
> > blocking: we have to update the so that it will using a single satellite for the upgrade tests.
> 
> On it

you need to write the session_target_sat fixture in the upgrade test.  

@pytest.fixture(scope='session')
def sat_gce(request, session_puppet_enabled_sat, session_target_sat):
    hosts = {'sat': session_target_sat}
    yield hosts[getattr(request, 'param', 'sat')]

and use `sat_gce` to all in upgrade tests.

File: tests/upgrades/test_host.py [hunk 4]
```diff
@@ -219,6 +195,6 @@ def test_post_create_gce_cr_and_host(
         ).create()
         request.addfinalizer(pre_upgrade_host.delete)
         request.addfinalizer(host.delete)
-        assert host.name == f"{self.hostname}-{domain}"
+        assert host.name == f"{self.hostname.lower()}.{domain.name}"
```
Comment (adarshdubey-star) [inline tests/upgrades/test_host.py:198]: @omkarkhatavkar I needed to change this one somehow the name was getting camel cased

File: tests/upgrades/test_host.py [hunk 5]
```diff
@@ -219,6 +195,6 @@ def test_post_create_gce_cr_and_host(
...
         assert host.build_status_label == 'Installed'
         assert host.ip == self.google_host(googleclient).ip
```
Comment (adarshdubey-star) [inline tests/upgrades/test_host.py:200]: Also at the end of this assertion the host is vanishing somehow

File: tests/upgrades/test_host.py (hunk 4 above)
Comment (omkarkhatavkar) [reply tests/upgrades/test_host.py:198]: okay make sense.

**Review by shubhamsg199 — APPROVED** [review]
Ack, Good work @adarshdubey-star

**Review by omkarkhatavkar — APPROVED** [review]

**Pull request merged**